crates/ruff_linter/src/rules/flake8_pyi/rules/redundant_numeric_union.rs

```diff
@@ -48,15 +62,23 @@ pub struct RedundantNumericUnion {
 }
 
 impl Violation for RedundantNumericUnion {
+    // Always fixable, but currently under preview.
+    const FIX_AVAILABILITY: FixAvailability = FixAvailability::Sometimes;
+
     #[derive_message_formats]
     fn message(&self) -> String {
         let (subtype, supertype) = match self.redundancy {
+            Redundancy::IntFloatComplex => ("int | float", "complex"),
             Redundancy::FloatComplex => ("float", "complex"),
             Redundancy::IntComplex => ("int", "complex"),
             Redundancy::IntFloat => ("int", "float"),
         };
         format!("Use `{supertype}` instead of `{subtype} | {supertype}`")
     }
+
+    fn fix_title(&self) -> Option<String> {
+        Some("Remove duplicates".to_string())
```

Comment:
Nit: To align with the terminology used in the documentation

```suggestion
        Some("Remove redundant type".to_string())
```